dev/src/main/java/org/eclipse/codewind/intellij/ui/tasks/ShowAllLogFilesTask.java

```diff
@@ -0,0 +1,235 @@
+        this.application = application;
+    }
+    @Override
+    public void run(@NotNull ProgressIndicator indicator) {
+        CoreUtil.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                initToolWindow();
+
+                for (ProjectLogInfo logInfo : application.getLogInfos()) {
+                    Content content = getContentForProjectLogInfo(logInfo);
+                    SocketConsole socketConsole = null;
+                    if (content == null) {
+                        socketConsole = createLogFileToolWindow(logInfo);
+                    }
+                    try {
+                        if (socketConsole != null) {
+                            socketConsole.initialize();
+                        }
+                    } catch( Exception e) {
+                        Logger.log(e);
+                    }
+                }
+                if (!logFilesToolWindow.isVisible() && logFilesToolWindow.isAvailable()) {
+                    logFilesToolWindow.show(null);
+                    logFilesToolWindow.activate(null, true, true);
+                }
+            }
+        });
+    }
+
+    private void initToolWindow() {
+        final ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(getProject());
+        logFilesToolWindow = toolWindowManager.getToolWindow(CodewindToolWindowHelper.SHOW_LOG_FILES_TOOLWINDOW_ID);
+        boolean windowRequiresRegistration = false;
+        if (logFilesToolWindow == null) {
+            windowRequiresRegistration = true;
+            logFilesToolWindow = toolWindowManager.registerToolWindow(CodewindToolWindowHelper.SHOW_LOG_FILES_TOOLWINDOW_ID, true, ToolWindowAnchor.BOTTOM);
+            logFilesToolWindow.setContentUiType(ToolWindowContentUiType.COMBO, null);
+            logFilesToolWindow.setIcon(IconCache.getCachedIcon(IconCache.ICONS_CODEWIND_13PX_SVG));
+        }
+        contentManager = logFilesToolWindow.getContentManager();
+        if (windowRequiresRegistration) {
+            contentManager.addContentManagerListener(new ContentManagerAdapter() {
+                public void contentRemoved(@NotNull ContentManagerEvent event) {
+                    ContentManagerEvent.ContentOperation operation = event.getOperation();
+                    if (operation.equals(ContentManagerEvent.ContentOperation.remove)) {
+
+                        String connectionId = application.connection.getConid();
+                        String projectId = application.projectID;
+                        if (connectionId != null && projectId != null) {
+                            CodewindToolWindow.getToolWindowUpdateHandler().removeAppUpdateListener(event.getContent(), connectionId, projectId);
+                        }
+
+                        Key<SocketConsole> key = CodewindToolWindow.getLogsKeyMap().get(event.getContent());
+                        SocketConsole sc = getProject().getUserData(key);
```

Comment:
IntelliJ indicates that `sc` might be null.  I don't know if there is a scenario where it can actually be null, but maybe it's worthwhile to add a null check?

![image](https://user-images.githubusercontent.com/29612372/75053861-fbd82080-549f-11ea-91da-db54f41d2e6b.png)

Reply:
Done